### DISCUSSION TITLE
[ZK Stack] - What are the roles and responsibilities of Validator Node in zksync?

### DISCUSSION THREAD

**Discussion by sachanayush47** [zkSync-Community-Hub/zksync-developers#849] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [X] Other (please specify below)

### Rust Version

1.82.0

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

We are trying to understand the roles and responsibilities of Validator Node aka External Node. As ZKSync, market itself as "secured by maths, not by validators" meaning Validator Node in ZKSync has a lot different responsibilities as compared with traditional Validator Node.

Our questions are:
1. What are the roles and responsibilities?
2. Does Validator Node increases the overall chain TPS?
3. What are the incentives of running Validator Node?

### Expected Behavior

NA

### Current Behavior

NA

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
### Roles and Responsibilities of a Validator Node in ZKsync

1. **Roles and Responsibilities:**
   - Validator Nodes in ZKsync are crucial for maintaining the network's integrity and security. They are responsible for producing and validating blocks, ensuring transactions are correctly processed and finalized. Unlike traditional blockchain systems where validators are primarily responsible for consensus, ZKsync leverages zero-knowledge proofs (ZKPs) which means the validators' role is more about ensuring the correctness of these proofs rather than achieving consensus through economic stakes.

2. **Impact on Chain TPS:**
   - Validator Nodes do not directly increase the overall transactions per second (TPS) of the chain. Instead, the scalability and speed of ZKsync are primarily achieved through its ZK-rollup technology, which batches multiple transactions into a single proof. The efficiency of the network is thus more reliant on the technology used rather than the number of validators.

3. **Incentives for Running a Validator Node:**
   - The incentives for running a Validator Node in ZKsync include earning transaction fees for processing transactions and creating new blocks. Additionally, maintaining a Validator Node contributes to the network's robustness and security, potentially increasing the node operator's reputation and influence within the ZKsync ecosystem.

For more detailed insights and discussions on ZKsync development, you can visit the [ZKsync developers discussions on Github](https://github.com/zkSync-Community-Hub/zksync-developers/discussions).

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by kaymomin** [comment]
Hey @sachanayush47, thanks for raising these ques! Let me loop in the right folks to provide clarity around the roles, responsibilities, and incentives of validator nodes soon.

**Reply by kaymomin** [reply, marked as answer]
So first, validator node and external node are different. An external node is what's normally called a full node on most blockchains. A validator node is a node that is involved in block production, something that we will have in the near future. 

**On the questions:**

1. The external node doesn't have any responsibilities. It's purpose is to connect to the main node and other external nodes in the network and receive blocks (and execute them). It allows the operator to know the full state of the blockchain. It really is just a full node.
2. No.
3. There's no incentives. People run external nodes if they need to have a copy of the blockchain state.

For more, check out the [docs here.](https://docs.zksync.io/zksync-era/tooling/external-node)
